cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -1040,6 +1040,63 @@ impl Inst {
         }
     }
 
+    /// Choose which instruction to use for comparing two values for equality.
```

Comment:
:+1: What do you think of putting it into a different impl of a (zero-side) struct called `InstHelper` (or dare I say, `MacroAssembler`)? It'll help make a clear distinction between the low-level assembler instructions represented by Vcode, vs the slightly higher level logic that's provided here.

Here and below: would it make sense to add non-SIMD types here, or put "simd" in the name if that's only achieved for SIMD instructions?

Reply:
I think my idea was to eventually add the scalar types here as well but the refactorings we talked about in #2252 are going to take a while so this is just the beginning.

Reply:
Um, after thinking about it a bit more I just stuffed the functions down in the `// Inst helpers.` block of the `impl Inst` so they still exist in `Inst`; I'll leave this PR up for you to take another look. I will add an `InstHelper` if you really think it would help but I figured creating a new struct would be more confusing at this point so we should either remove these functions or discuss #2252 a bit more before refactoring. (We can always move them later--and `Inst::load` and `Inst::store` are probably in the same class).